shared/js/background/events/referrer-trimming.js

```diff
@@ -3,18 +3,24 @@ const trackerutils = require('../tracker-utils')
 const utils = require('../utils.es6')
 const browserName = utils.getBrowserName()
 
+/**
+ * @typedef Referrer
+ * @property {string} site
+ *   The referrer URL.
+ * @property {string} referrerHost
+ *   The referrer host.
+ * @property {string} referrer
+ *   The truncated referrer.
+ */
```

Comment:
Not a bug; just a suggestion we could discuss, if you differ.

I think we should prefer classes where possible like:
```suggestion
 class Referrer {
     /**
      * @property {string} site The referrer URL.
      * @property {string} referrerHost The referrer host.
      * @property {string} referrer The truncated referrer.
      */
     constructor(site, referrerHost, referrer) {
         this.site = site
         this.referrerHost = referrerHost
         this.referrer = referrer
     }
 }
```

This means the usage down below would become typed too with:
```
tab.referrer = new Referrer(tab.site.url, new URL(referrer).hostname, modifiedReferrer)
```

- The consumer site in tab would then import a class, in some cases this also has the benefit that it can be a regular import in JS rather than a TS comment.
- I also like this model as it can serve as an extension point for anything that this object represents, the concerns can be methods instead.